src/hpc/cpu.c

```diff
@@ -0,0 +1,74 @@
+/****************************************************************************
+**
+**  This file is part of GAP, a system for computational discrete algebra.
+**
+**  Copyright of GAP belongs to its developers, whose names are too numerous
+**  to list here. Please refer to the COPYRIGHT file for details.
+**
+**  SPDX-License-Identifier: GPL-2.0-or-later
+*/
+
+#include "hpc/cpu.h"
+
+/****************************************************************************
+**
+*V  SyNumProcessors  . . . . . . . . . . . . . . . . . number of logical CPUs
+**
+*/
+#ifdef NUM_CPUS
+UInt SyNumProcessors = NUM_CPUS;
```

Comment:
I wonder: should we even bother to keep `NUM_CPUS`? Your code kinda renders it obsolete, no? the user can still override it at runtime using a command line option anyway. What do you think?
 
@ChrisJefferson any thoughts on that?

Reply:
> the user can still override it at runtime using a command line option anyway

Didn't know about this option. Seems like `NUM_CPUS` is not needed then.
Removed it in 7d43aa767c9a30b85c9424d2487171f327b1cc13. If it is needed, I can revert it.

Reply:
Yes, I don't think we need this NUM_CPUS (just agreeing)